cmd/queue/main.go

```diff
@@ -277,7 +305,13 @@ func probeQueueHealthPath(port int, timeoutSeconds int) error {
 			return false, nil
 		}
 		defer res.Body.Close()
-		return health.IsHTTPProbeReady(res), nil
+		success := health.IsHTTPProbeReady(res)
+		// The check for preferForScaledown() fails readiness faster
+		// in the presence of the label
+		if !success && preferPodForScaledown(logger, env) {
+			cancel()
```

Comment:
This cancel seems unneeded. Does `return health.IsHTTPProbeReady(res) || preferPodForScaledown(logger, env)` yield the same semantics? Or even

```
if !success && preferPodForScaledown(logger, env) {
  return false, errors.New("failing probe deliberately for pod scaledown")
}
```

To make the intention more clear? Cancelling the context will I guess result in a timeout error but that feels a little backwards.

Reply:
yeah had it return `false` and an error initially but `cancel` felt crispier. will switch to your suggested change.

Reply:
even though this feels more like an issue with `PollImmediateUntil` since `cancel` should imply breaking out of the loop without success. nothing to do with timeout but well ...  ¯\_(ツ)_/¯